db/schema_tables.cc

```diff
@@ -330,12 +330,13 @@ schema_ptr tables() {
 
 // Holds Scylla-specific table metadata.
 schema_ptr scylla_tables(schema_features features) {
-    static thread_local schema_ptr schemas[2][2]{};
```

Comment:
Wouldn't it be simpler to leave 'version' alone and add 'group0_version' instead of the flag?

Reply:
I don't think it would be simpler. Thanks to using `version`, I'm also reusing a lot of pre-existing code and structures.

First, the mutations that we're sending in `announce` are already containing cells for the `version` column. It's just that we're currently removing these cells when applying schema. My patch is to not remove these cells if we know that the schema comes from group 0.

Furthermore, when building schema objects, the existing code is already checking if `version` column is not null, and if not, it uses that value instead of calculating a digest. So thanks to using the `version` column, most of the issue is solved automagically!

With `group0_version` column I would duplicate the information in mutation already passed in `version` column. Furthermore I would either have to modify existing code or more likely add new code (so if something depends on existing code, it doesn't break) to load data from this `group0_version` column instead of `version`. It's generally more work, the patchset would be larger.

Reply:
> First, the mutations that we're sending in announce are already containing cells for the version column. It's just that we're currently removing these cells when applying schema. My patch is to not remove these cells if we know that the schema comes from group 0.

But we need to add another cell to the mutation so this is not really a benefit.

> Furthermore, when building schema objects, the existing code is already checking if version column is not null, and if not, it uses that value instead of calculating a digest. So thanks to using the version column, most of the issue is solved automagically!

Yes, so I was thinking that instead of having the code that checks if 'version' needs to be removed we will have a code that checks `version` and `group0_version`. But if you think the complexity is similar then there is no difference.